kernel/src/expressions/scalars.rs

```diff
@@ -92,6 +109,16 @@ impl Display for Scalar {
                 }
             },
             Self::Null(_) => write!(f, "null"),
+            Self::Struct(values, fields) => {
+                write!(f, "{{")?;
+                for (i, (value, field)) in values.iter().zip(fields.iter()).enumerate() {
+                    if i > 0 {
+                        write!(f, ", ")?;
+                    }
+                    write!(f, "{}: {}", field.name, value)?;
```

Comment:
tiny nit: can also use this trick for comma separators:
```rust
let mut delim = "";
for (...) {
  write!(f, "{delim}{}: {value}", field.name)?;
  delim = ", ";
}
```
(in perf sensitive code, it avoids branching; here it just avoids the need to `enumerate`)

Reply:
... and a few lines :).